core/tee/tee_svc_storage.c

```diff
@@ -307,6 +307,12 @@ static TEE_Result tee_svc_storage_read_head(struct tee_ta_session *sess,
 
 	fops = o->pobj->fops;
 
+	o->pobj->head = malloc(sizeof(struct tee_svc_storage_head));
```

Comment:
Whoever updated o->pobj->head would cause it to have the latest copy? As
long as we keep 1 single malloc for the object.

On Thu, Jul 14, 2016 at 7:23 PM, GuanchaoLiang <notifications@github.com>
wrote:

> In core/tee/tee_svc_storage.c
> <https://github.com/OP-TEE/optee_os/pull/912#discussion_r70913590>:
>
> > @@ -307,6 +307,12 @@ static TEE_Result tee_svc_storage_read_head(struct tee_ta_session *sess,
> >
> >  	fops = o->pobj->fops;
> >
> > +	o->pobj->head = malloc(sizeof(struct tee_svc_storage_head));
>
> But I still find a problem that, if A open the obj, and then B open the
> obj and write, after that A write, the condition seems that A will update
> the head without B's update?
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/OP-TEE/optee_os/pull/912/files/b95f28948850c1876c98d24acd925f77bbfdfde9#r70913590>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AAR8hy4b2qnO89WINVSxTSZk9xALsrZBks5qVu8VgaJpZM4JKg_J>
> .
>